### PR TITLE
Vulkan Window Surface Creation

### PR DESCRIPTION
**Pull request by Rinzii**

### CODE REVIEW

**Review by Kouros26 — APPROVED** [review]

File: engine/src/graphics/device.cpp [hunk 1]
```diff
@@ -15,16 +15,18 @@
...
 		// TODO: Allow for the vulkan api to be specified in a config file or through command line arguments.
 		// We could also dynamically check for the highest supported version of the vulkan api, but that feels outside the scope of this project.
 		createInstance(appName, "Genesis Engine", VK_API_VERSION_1_3);
+		createSurface(window);
 	}
 
 	GraphicsDevice::~GraphicsDevice()
 	{
 
+		m_surface.release();
```
Comment (karnkaul) [inline engine/src/graphics/device.cpp]: Why are we doing this manually though? Just having a default destructor and the correct member order will take care of it automatically. Kinda defeats the purpose of using RAII types.

**Review by karnkaul — CHANGES_REQUESTED** [review]

File: engine/src/graphics/device.cpp [hunk 2]
```diff
@@ -26,7 +26,6 @@ namespace gen
 	GraphicsDevice::~GraphicsDevice()
 	{
 
-		m_surface.release();
 		m_instance.release();
```
Comment (karnkaul) [inline engine/src/graphics/device.cpp]: Now instance gets destroyed before surface 😄

**Review by karnkaul — CHANGES_REQUESTED** [review]
Note: vkUtils [still adds](https://github.com/Rinzii/genesis/blob/8e80e62698b3a5a39688e165e46cdac311f52807/engine/src/graphics/vkHelpers.cpp#L62) `VK_EXT_DEBUG_UTILS_EXTENSION_NAME` into the list of instance extensions.

File: engine/include/graphics/device.hpp [hunk 3]
```diff
@@ -20,12 +20,15 @@ namespace gen
 		GraphicsDevice(GraphicsDevice &&)				   = delete;
 		GraphicsDevice & operator=(GraphicsDevice &&)	   = delete;
 
-		explicit GraphicsDevice( std::string const & appName );
-		~GraphicsDevice();
+		explicit GraphicsDevice(Window &window, std::string const & appName );
```
Comment (karnkaul) [inline engine/include/graphics/device.hpp]: `window` can be `const&`, can't it?

**Review by karnkaul — APPROVED** [review]

**Pull request merged**